typescript/commands/references.py

```diff
@@ -173,3 +191,42 @@ def run(self, text, argsJson):
         # serialize the reference info into the settings
         self.view.settings().set('refinfo', ref_info.as_value())
         self.view.set_read_only(True)
+
+class RefsHelper:
+
+    __lines = {}
+    __content = {}
+
+    def lineText(self, fileName, line):
+        """ returns text of the given line in the given file """
+        lines = self.__table(fileName)
+        l = len(lines)
+        if l < line:
+            return ''
+        content = self.__content[fileName]
+        start = lines[line]
+        if line < l - 1:
+            end = lines[line + 1]
+        else:
+            end = len(content)
+        return content[start:end].rstrip()
+
+    def __table(self, fileName):
+        """ returns line table for a given filename, computes if needed """
```

Comment:
> This should be in init, right? Any reason to do the initialization lazily?

Nope, I think that we should initialize line offset tables lazily because purpose of this class is to transform pairs `(file F, line L)` into `text of line L in file F`. I see no reason to iterate over all references and extract unique file names before asking for line's text just to move this code to `init`